### PR TITLE
Create a single helper for the various versions of addAnnotation and addLabel

### PR DESCRIPTION
**Pull request by ELENAGER**
The Ramen codebase currently contains several versions of AddAnnotation() and AddLabel() functions. This PR changes it to use functions in util/misc.go and cleans up all other occurances.

Fixes: https://github.com/RamenDR/ramen/issues/1097

### CODE REVIEW

**Review by ShyamsundarR — APPROVED** [review]
LGTM! Will wait for one more review before merge.

**Review by raghavendra-talur — APPROVED** [review]

**Pull request merged**
